### PR TITLE
ci: remove concurrency tag. Merge test projects and use redis locker

### PR DESCRIPTION
**Pull request by Matovidlo**
Jira: [PSGO-528](https://keboola.atlassian.net/browse/PSGO-528)

**Changes:**
- remove `concurrency` tag from pipeline.
- merge `$TEST_KBC_$PLATFORM_PROJECTS` together into one variable `$TEST_KBC_PROJECTS`.
- move the `$TEST_KBC_PROJECTS` into github actions repository environment variables. 
- add `envsubst` step for substitution of secrets into new `$TEST_KBC_PROJECTS` variable back.

-----------

[PSGO-528]: https://keboola.atlassian.net/browse/PSGO-528?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/test-e2e-cli.yml [hunk 1]
```diff
@@ -9,75 +9,35 @@ on:
 
 env:
   GO_VERSION: 1.22.2
-  TEST_PARALLELISM: 8
+  TEST_PARALLELISM: 16
   TEST_PARALLELISM_PKG: 1
```
Comment (Matovidlo) [inline .github/workflows/test-e2e-cli.yml:13]: Should we also change this? Is there any reason why e2e cli tests were running maximum 1 per package? Probably because it is a single package.

File: .github/workflows/test-unit.yml [hunk 2]
```diff
@@ -5,63 +5,35 @@ on:
 
 env:
   GO_VERSION: 1.22.2
-  TEST_PARALLELISM: 6
+  TEST_PARALLELISM: 12
   TEST_PARALLELISM_PKG: 6
```
Comment (Matovidlo) [inline .github/workflows/test-unit.yml:9]: Should we increase paralellism of unit test packages? Looks like the tests are not running faster.

File: .github/actions/export-kbc-projects/action.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+  # Temporary removed project: https://keboolaglobal.slack.com/archives/C055HSMKX51/p1714024266416349
+#  { "host": "connection.north-europe.azure.keboola.com","project": 10890,"stagingStorage": "abs","backend": "snowflake","token": "$TEST_KBC_PROJECT_10890_TOKEN", "legacyTransformation": true },
+#  { "host": "connection.north-europe.azure.keboola.com","project": 10891,"stagingStorage": "abs","backend": "snowflake","token": "$TEST_KBC_PROJECT_10891_TOKEN", "legacyTransformation": true }
+#  { "host":"connection.north-europe.azure.keboola.com","project":10892,"stagingStorage":"abs","backend":"snowflake","token":"$TEST_KBC_PROJECT_10892_TOKEN", "legacyTransformation": true}
+#  {"host":"connection.north-europe.azure.keboola.com","project":10898,"stagingStorage":"abs","backend":"snowflake","token":"$TEST_KBC_PROJECT_10898_TOKEN", "legacyTransformation": true},
```
Comment (Matovidlo) [inline .github/actions/export-kbc-projects/action.yml:5]: Should we remove these projects? We have new ones, but the KBC team did fixes so the `workspace not found` error was probably erradicated from the tests results.

File: .github/workflows/test-e2e-cli.yml (hunk 1 above)
Comment (michaljurecko) [reply .github/workflows/test-e2e-cli.yml:13]: Yes, there is only one package. 

Yes, you should increase `TEST_PARALLELISM`.

File: .github/actions/export-kbc-projects/action.yml (hunk 3 above)
Comment (michaljurecko) [reply .github/actions/export-kbc-projects/action.yml:5]: You can enable them again, if they work, maybe some projects are already deleted cc @hosekpeter

Comment (michaljurecko) [reply .github/actions/export-kbc-projects/action.yml:5]: I don't know if the workspace delete operation is fixed, or the error doesn't occure more.

File: .github/workflows/test-e2e-cli.yml [hunk 4]
```diff
@@ -9,75 +9,35 @@ on:
...
-
+      
       - name: Checkout code
         uses: actions/checkout@v4
 
+      - name: Set TEST_KBC_PROJECTS
+        uses: ./.github/actions/export-kbc-projects
+        with:
+          secrets: ${{ toJSON(secrets) }}
+          projects: ${{ vars.TEST_KBC_PROJECTS }}
```
Comment (michaljurecko) [inline .github/workflows/test-e2e-cli.yml:39]: This inputs are same in all cases, so maybe we can remove them (?).

File: .github/workflows/test-e2e-cli.yml [hunk 5]
```diff
@@ -98,4 +58,6 @@ jobs:
         shell: bash
         env:
           VERSION: ${{ inputs.version }}
-          TEST_KBC_PROJECTS: ${{ env[matrix.projects] }}
+          TEST_KBC_PROJECTS_LOCK_HOST: ${{ secrets.TEST_KBC_PROJECTS_LOCK_HOST }}
```
Comment (michaljurecko) [inline .github/workflows/test-e2e-cli.yml:61]: Host should be variable not secret, now, I dont know which Redis is CI using (sorry, I didn't mentioned it in the previous PR).

File: .github/workflows/test-e2e-service-stream.yml [hunk 6]
```diff
@@ -5,17 +5,8 @@ on:
 
 env:
   GO_VERSION: 1.22.2
-  TEST_PARALLELISM: 4
+  TEST_PARALLELISM: 8
```
Comment (michaljurecko) [inline .github/workflows/test-e2e-service-stream.yml:8]: `16`? Or we can wait and increase it more, when test caching will be ready.

File: .github/workflows/test-unit.yml (hunk 2 above)
Comment (michaljurecko) [reply .github/workflows/test-unit.yml:9]: Unit test are using testing project in a small number of tests, CLI E2E tests use a testing project in each test.

File: .github/workflows/test-e2e-cli.yml (hunk 1 above)
Comment (Matovidlo) [reply .github/workflows/test-e2e-cli.yml:13]: It is done

File: .github/workflows/test-unit.yml (hunk 2 above)
Comment (Matovidlo) [reply .github/workflows/test-unit.yml:9]: ok. So I just increased the `TEST_PARALLELISM`: 6 -> 12. `TEST_PARALLELISM_PKG` will remain 6.

File: .github/actions/export-kbc-projects/action.yml (hunk 3 above)
Comment (Matovidlo) [reply .github/actions/export-kbc-projects/action.yml:5]: I'll try it and we will see. If problem remains i will delete them forewer with secrets.

File: .github/workflows/test-e2e-cli.yml (hunk 4 above)
Comment (Matovidlo) [reply .github/workflows/test-e2e-cli.yml:39]: When you have reusable workflow which we do, we have to have `inputs` field where we insert these variables. I had it without any of these and the reusable workflow did not see these variables and secrets

File: .github/workflows/test-e2e-cli.yml (hunk 5 above)
Comment (Matovidlo) [reply .github/workflows/test-e2e-cli.yml:61]: Will rework both.

File: .github/workflows/test-e2e-service-stream.yml (hunk 6 above)
Comment (Matovidlo) [reply .github/workflows/test-e2e-service-stream.yml:8]: We can increase it so at least we see how much it is running faster.

File: .github/workflows/test-e2e-cli.yml (hunk 4 above)
Comment (michaljurecko) [reply .github/workflows/test-e2e-cli.yml:39]: Ok, ... there is also alternative `secrets: inherit`

Comment (Matovidlo) [reply .github/workflows/test-e2e-cli.yml:39]: As we have discussed we can leave it like that. It is the simillar to the `secrets: inherit`

**Comment by Matovidlo** [comment]
waiting for https://github.com/keboola/go-utils/pull/33 to be merged and new tag created.

**Review by michaljurecko — APPROVED** [review]
LGTM
